security-proxy/src/main/java/org/georchestra/security/Proxy.java

```diff
@@ -364,12 +366,14 @@ public void handleDefaultRequest(HttpServletRequest request, HttpServletResponse
      */
     @GetMapping(value = "/whoami", produces = MediaType.APPLICATION_JSON_VALUE)
     @ResponseBody
-    public String whoami(HttpServletRequest request) throws DataServiceException {
+    public ResponseEntity<String> whoami(HttpServletRequest request) throws DataServiceException {
```

Comment:
I think if you return the JSONObject directly instead of a string representation the encoding will be properly handled by GetMapping with produces = MediaType.APPLICATION_JSON_VALUE?

Reply:
i've tried it (returning directly the JSONObject), and spring only returned 406 codes, whatever the accept header i manually set (and we cant change all the consumers of the API to set that header)

Reply:
> I think if you return the JSONObject directly instead of a string representation the encoding will be properly handled by GetMapping with produces = MediaType.APPLICATION_JSON_VALUE?

i know i've tried that, but i've read somewhere that `produces` didnt do what we expect (eg from my understanding it was more 'sanity check the type of the value' and **not** 'set the `content-type` header' -  in the end i mostly tried the examples from https://www.baeldung.com/spring-mvc-set-json-content-type.

looking at spring-projects/spring-framework#19050 there's a `@DefaultContentType` annotation ?

maybe that's due to the fact the S-P uses spring 4.3.22 which is somewhat old (2019) ?

anyway, the S-P is on life support, so maybe not worth spending too much time on this.